lib/modules/dosomething/dosomething_helpers/dosomething_helpers.module

```diff
@@ -576,8 +576,8 @@ function dosomething_helpers_node_metadata_wrapper($node, $ignore_language = FAL
  *   ISO-code of current language
  */
 function dosomething_helpers_get_current_language_code() {
-  global $language;
-  return $language->language;
+  global $language_content;
```

Comment:
This should be right since we're detecting language for field content. On the other hand, somebody could use this helper for other purposes. So you are right. I'll move this to its own function.